ocaml/xapi/repository_helpers.ml

```diff
@@ -1478,3 +1524,114 @@ let set_pending_guidances ~ops ~coming =
            to_be_removed ;
          do_with_vm_pending_guidances ~op:ops.vm_add ~vm:vm_ref_str to_be_added
      )
+
+let failure_of_livepatch_component = function
```

Comment:
Just to confirm, the `component` field of livepatch in updateinfo exists for all the updates: both the old ones and the new ones to be generated?

Reply:
If we can always determine a livepatch failure to be `RebootHostOnXenLivePatchFailure` or `RebootHostOnKernelLivePatchFailure`, I think we can remove the handling for guidance: `reboot_host_on_livepatch_failure`, as any remaining ones should have been replaced to both `reboot_host_on_xen_livepatch_failure` and `reboot_host_on_kernel_livepatch_failure` by the added db upgrade rule during xapi startup.

Reply:
The type variable is still defined there. I think it's harmless here and it's safe to list all failures here explicitly.

Reply:
I think the type `reboot_host_on_livepatch_failure` is not needed anymore?

Reply:
Yes, you are right, the type `reboot_host_on_livepatch_failure` is still needed for that livepatch failure guidance "db upgrade rule".